### PR TITLE
Include rabbitmq, MSMQ, ASQ and ASB

### PR DESCRIPTION
**Pull request by WilliamBZA**
Connects to: https://github.com/Particular/V6Launch/issues/79

Dependent on: https://github.com/Particular/EndToEnd/pull/148

Adds MSMQ and RabbitMQ transports to the test categories.

Ping @HEskandari @mikeminutillo @ramonsmits @gbiellem

### CODE REVIEW

**Comment by WilliamBZA** [comment]
Added the other transports too

File: src/PlatformCompatibilityTests/Sql/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (HEskandari) [inline src/PlatformCompatibilityTests/Sql/Properties/AssemblyInfo.cs]: Errr....I don't think this is the right AssemblyInfo. Probably need to copy one from other projects with the right Copyright/Company/etc.

**Comment by HEskandari** [comment]
Just a side note that I don't see TeamCity build status on the PRs. Something we need to setup like we did for NHibernate Compat testing maybe.

File: src/PlatformCompatibilityTests/ASQ/ASQTransportDetails.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace ServiceControlCompatibilityTests
+{
+    public class ASQTransportDetails : ITransportDetails
+    {
+        const string TransportTypeName = "NServiceBus.AzureStorageQueueTransport, NServiceBus.Azure.Transports.WindowsAzureStorageQueues";
+
+        public ASQTransportDetails(string connectionString)
+        {
+            connectionString = "UseDevelopmentStorage=true";
```
Comment (mikeminutillo) [inline src/PlatformCompatibilityTests/ASQ/ASQTransportDetails.cs]: Should this be a `+=` or something? Why are we overwriting what gets passed in?

File: src/PlatformCompatibilityTests/ServiceControlCompatibilityTests/Infrastructure/Msmq/MsmqTransportDetails.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+        public void ApplyTo(Configuration configuration)
+        {
+            var settings = configuration.AppSettings.Settings;
+            settings.Set(SettingsList.TransportType, TransportTypeName);
+        }
+
+        public void ConfigureEndpoint(EndpointConfiguration endpointConfig)
+        {
+            endpointConfig.PurgeOnStartup(true);
```
Comment (mikeminutillo) [inline src/PlatformCompatibilityTests/ServiceControlCompatibilityTests/Infrastructure/Msmq/MsmqTransportDetails.cs]: Should we rely on the defaults here or explicitly set the transport to MSMQ?

**Comment by mikeminutillo** [comment]
Overall it looks good and supporting those other transports is surprisingly painless.

File: src/PlatformCompatibilityTests/ASQ/ASQTransportDetails.cs (hunk 2 above)
Comment (WilliamBZA) [inline src/PlatformCompatibilityTests/ASQ/ASQTransportDetails.cs]: Uhh oops, that shouldn't have made it in...

File: src/PlatformCompatibilityTests/ServiceControlCompatibilityTests/Infrastructure/Msmq/MsmqTransportDetails.cs (hunk 3 above)
Comment (WilliamBZA) [inline src/PlatformCompatibilityTests/ServiceControlCompatibilityTests/Infrastructure/Msmq/MsmqTransportDetails.cs]: Defaults should be fine. We could be explicit about it for clarity. Don't think there'd be much more benefit than that.

**Pull request merged**